esphome/components/micronova/micronova.h

```diff
@@ -0,0 +1,164 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/components/uart/uart.h"
+#include "esphome/core/log.h"
+#include "esphome/core/defines.h"
+#include "esphome/core/helpers.h"
+
+#include <vector>
+
+namespace esphome {
+namespace micronova {
+
+static const char *const TAG = "micronova";
+static const int STOVE_REPLY_DELAY = 60;
+
+static const std::string STOVE_STATES[11] = {"Off",
+                                             "Start",
+                                             "Pellets loading",
+                                             "Igniton",
+                                             "Working",
+                                             "Brazier Cleaning",
+                                             "Final Cleaning",
+                                             "Standby",
+                                             "No pellets alarm",
+                                             "No ignition alarm",
+                                             "Undefined alarm"};
+
+enum class MicroNovaFunctions {
+  STOVE_FUNCTION_VOID = 0,
+  STOVE_FUNCTION_SWITCH = 1,
+  STOVE_FUNCTION_ROOM_TEMPERATURE = 2,
+  STOVE_FUNCTION_THERMOSTAT_TEMPERATURE = 3,
+  STOVE_FUNCTION_FUMES_TEMPERATURE = 4,
+  STOVE_FUNCTION_STOVE_POWER = 5,
+  STOVE_FUNCTION_FAN_SPEED = 6,
+  STOVE_FUNCTION_STOVE_STATE = 7,
+  STOVE_FUNCTION_MEMORY_ADDRESS_SENSOR = 8,
+  STOVE_FUNCTION_WATER_TEMPERATURE = 9,
+  STOVE_FUNCTION_WATER_PRESSURE = 10,
+  STOVE_FUNCTION_POWER_LEVEL = 11,
+  STOVE_FUNCTION_CUSTOM = 12
+};
+
+class MicroNova;
+
+//////////////////////////////////////////////////////////////////////
+// Interface classes.
+class MicroNovaBaseListener {
+ public:
+  MicroNovaBaseListener() {}
+  MicroNovaBaseListener(MicroNova *m) { micronova_ = m; }
+  virtual void dump_config();
+
+  void set_micronova_object(MicroNova *m) { micronova_ = m; }
+
+  void set_function(MicroNovaFunctions f) { function_ = f; }
+  MicroNovaFunctions get_function() { return function_; }
+
+  void set_memory_location(uint8_t l) { memory_location_ = l; }
+  uint8_t get_memory_location() { return memory_location_; }
+
+  void set_memory_address(uint8_t a) { memory_address_ = a; }
+  uint8_t get_memory_address() { return memory_address_; }
+
+ protected:
+  MicroNova *micronova_{nullptr};
+  MicroNovaFunctions function_ = MicroNovaFunctions::STOVE_FUNCTION_VOID;
+  uint8_t memory_location_ = 0;
+  uint8_t memory_address_ = 0;
+};
+
+class MicroNovaSensorListener : public MicroNovaBaseListener {
+ public:
+  MicroNovaSensorListener() {}
+  MicroNovaSensorListener(MicroNova *m) : MicroNovaBaseListener(m) {}
+  virtual void request_value_from_stove();
+  virtual void process_value_from_stove(int value_from_stove);
```

Comment:
Should these be changed to enforce implementation in the final class?
```suggestion
  virtual void request_value_from_stove() = 0;
  virtual void process_value_from_stove(int value_from_stove) = 0;
```

Reply:
Correct, done.